core/src/main/java/org/mule/transformer/simple/AbstractAddVariablePropertyTransformer.java

```diff
@@ -68,17 +67,21 @@ public Object transformMessage(MuleEvent event, String outputEncoding) throws Tr
                 {
                     DataType<?> dataType = DataTypeFactory.create(typedValue.getValue().getClass(), getMimeType());
                     dataType.setEncoding(getEncoding());
-                    event.getMessage().setProperty(key, typedValue.getValue(), getScope(), dataType);
+                    addProperty(event, key, typedValue.getValue(), dataType);
                 }
                 else
                 {
-                    event.getMessage().setProperty(key, typedValue.getValue(), getScope(), typedValue.getDataType());
+                    addProperty(event, key, typedValue.getValue(), typedValue.getDataType());
                 }
             }
         }
         return event.getMessage();
     }
 
+    protected abstract void addProperty(MuleEvent event, String propertyName, Object value, DataType dataType);
```

Comment:
Would be good to have javadoc on abstract methods to clearly indicate what they are supposed to do